### PR TITLE
Adds Mr. Yuma to the waste

### LINKED ISSUE #278: alert factions when a member matrixes
alert factions when members use the matrix, so people dont go looking for people that wordlessly matrix

### PR DESCRIPTION
**Pull request by xTrainx**
## About The Pull Request
Adds an announcement system: Mr. Yuma, that can announce when someone is matrix'ed, to the ~~common channel~~ The faction c hannel :D, aswell as the normal functions of an announcement system. Closes #278 

## Pre-Merge Checklist
- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

## Changelog
:cl:
add: adds Mr. Yuma announcement system to the train station
tweak: other factions won't get to claim town at roundstart, will need to massacer the people the good old fashioned way
/:cl:

### CODE REVIEW

File: code/modules/fallout/turf/walls.dm [hunk 1]
```diff
@@ -324,8 +324,16 @@ turf/closed/wall/f13/wood/house/update_damage_overlay()
 	log_admin(dat)
 	if(departing_mob.stat == DEAD)
 		departing_mob.visible_message(span_notice("[user] pushes the body of [departing_mob] over the border. They're someone else's problem now."))
+		if(GLOB.announcement_systems.len)
+			var/obj/machinery/announcement_system/announcer = pick(GLOB.announcement_systems)
+			announcer.announce("FACTIONLEAVEDEAD", departing_mob.real_name, departing_mob.job, list())
+			message_admins("sent the announcement")
```
Comment (RonBag) [inline code/modules/fallout/turf/walls.dm:330]: Look like debug logs left over

Comment (RonBag) [inline code/modules/fallout/turf/walls.dm:330]: Looks like debug logs left over

File: code/modules/fallout/turf/walls.dm [hunk 2]
```diff
@@ -324,8 +324,16 @@ turf/closed/wall/f13/wood/house/update_damage_overlay()
...
 	else
 		departing_mob.visible_message(span_notice("[departing_mob == user ? "Out of their own volition, " : "Ushered by [user], "][departing_mob] crosses the border and departs the area."))
+		if(GLOB.announcement_systems.len)
+			var/obj/machinery/announcement_system/announcer = pick(GLOB.announcement_systems)
+			announcer.announce("FACTIONLEAVE", departing_mob.real_name, departing_mob.job, list())
+			message_admins("sent the announcement")
```
Comment (RonBag) [inline code/modules/fallout/turf/walls.dm:336]: Same here

File: code/modules/fallout/turf/walls.dm (hunk 1 above)
Comment (xTrainx) [reply code/modules/fallout/turf/walls.dm]: oops

File: code/modules/fallout/turf/walls.dm (hunk 2 above)
Comment (xTrainx) [reply code/modules/fallout/turf/walls.dm]: 😳

File: code/game/machinery/announcement_system.dm [hunk 3]
```diff
@@ -21,23 +21,55 @@ GLOBAL_LIST_EMPTY(announcement_systems)
...
 	var/newheadToggle = 1
+	var/factionleave = "%PERSON, the %RANK, has left these lands for a (hopefully) better land in the wasteland." //this is used to announce a faction of when a player leaves by the matrix
+	var/factionleavedead = "the corpse of %PERSON, the %RANK, is now feeding the vultures, at the other side of the border" //same but dead
+
 
 	var/greenlight = "Light_Green"
 	var/pinklight = "Light_Pink"
 	var/errorlight = "Error_Red"
 
+/obj/machinery/announcement_system/Mr_Yuma //ATOM EDIT -Start- , adds a lore friendly announcer in the form of Mr. Yuma, a parody of Mr. New Vegas
```
Comment (RonBag) [inline code/game/machinery/announcement_system.dm:32]: set `arrivalToggle = FALSE` as right now it's announcing everyone arriving over common.

**Review by RonBag — CHANGES_REQUESTED** [review]

File: code/game/machinery/announcement_system.dm (hunk 3 above)
Comment (xTrainx) [reply code/game/machinery/announcement_system.dm:32]: rightto, done

**Review by RonBag — APPROVED** [review]

**Pull request closed without merging**